### ISSUE TITLE
[Elemental Shaman] Add Basic Support for this Spec

### ISSUE THREAD

**Issue by fasib** [MartijnHols/WoWAnalyzer#51]
- [ ] Update templates to support DPS specs
- [x] Add option for multiple talent descriptions (e.g. <a href="http://www.wowhead.com/spell=192063" target="_blank">Gust of Wind</a> description for ele and restro) #50
- [ ] Add all shaman spells to [SPELLS_SHAMAN.js](https://github.com/MartijnHols/WoWAnalyzer/blob/master/src/common/SPELLS_SHAMAN.js)
- [ ] Detect Maelstrom usage and generation

**Comment by MartijnHols** [comment]
@fasib you're working on adding this, right?

**Comment by fasib** [comment]
@MartijnHols yeah I'm working on it, I will keep this issue up to date with the progress.

**Comment by Sophie2712** [comment]
The message about going oom and failing to use a mana pot during a fight for ele shamans is a bit redundant, since we use maelstrom for the majority of, if not all, of our rotation. I'm not sure I've ever oom'ed since this change, but it says I have

**Comment by Yuyz0112** [comment]
@Sophie2712 
Also meet this bug and give a fix to it in #118

**Comment by HawkCorrigan** [comment]
Just a quick question/bug report
For my log and the 3/4 i crosschecked totem mastery uptime is shown as 0%
That seems to happen because you only take the minimum value for the calculation and StormTotemUptime is 0.
For some reason the Storm Totem Buff isn't even shown in warcraftlogs so I'd guess that this is the problem

It might be benefitial to just exclude the StormTotem from the check if it's not fixable from this side.

**Comment by janvavra** [comment]
Hi @fasib, any plans for the future? I have some development time and might be able to help with some tasks. It would be best to coordinate. 
What I feel is most needed right now for Antorus:

- implement Elementalist (Elemental CD reduction with LavaBurst) to get true Elemental utilization
- common mistakes in rotations for all specs (based on Storm, Earth and Lava):
  - Lava Burst should always crit
  - Lighting Bolt during Ascendance  
  - wasting Stormkeeper procs 
  - refreshing Flame Shock without Elemental Focus (minor)   
- move to separating Icefury and Gambling builds out to check for their specific rotational mistakes

**Comment by fasib** [comment]
Hi @janvavra, thanks for the help. I didn't have so much time in the past months and just started a few days ago to refactor some not yet commited code into the analyzer (I already implemented some rotation mistake checks).

- Elementalist could get interesting without some changes to the Core CastEfficiency Module (which right now only supports haste modifiers)
- A basic Elementalis Module with a Statsitic Box should be a good start for the first Module :)

I will cleanup the TODO's and write down my plans later today.

You can contact me @Discord

-----------------

@HawkCorrigan yep, fixed it finaly in #635 ...
